vocode/streaming/models/transcript.py

```diff
@@ -213,3 +216,14 @@ def to_string(self, include_timestamp: bool = False) -> str:
 
 class TranscriptCompleteEvent(Event, type=EventType.TRANSCRIPT_COMPLETE):
     transcript: Transcript
+
+
+class StateAgentTranscriptEvent(Event, type=EventType.STATE_AGENT_PUBLISH):
+    transcript: StateAgentTranscript
+
+    @classmethod
```

Comment:
what's the tradeoff between this and a constructor?

Reply:
because there is some extra validation it does ,you can also see this in the code style for the pydantic event , which does the same thing

Reply:
my follow up question is: do we need a constructor or @classmethod at all? It looks like the other events are instantiated with a constructor, but they don't need to override it

> because there is some extra validation it does

what does it do?

Reply:
Yes we do, I need the deepcopy otherwise with the async stuff it will be undepredicatable when we send the obj we might send the latest one.
Classmethod is the convention for this kind of factory function, or we can or do staticmethod both would look very similar.
The whole point of pydantic is we get to add some custom validation (like check this is some int or do some kind of custom conversion, or check the int is in some range or something like that, we may not use this now but this is very typical behaviour and the default case, so I would like to not break it) and serialization to the library.